includes/embeds/class-amp-youtube-embed-handler.php

```diff
@@ -77,6 +80,90 @@ public function unregister_embed() {
 		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10 );
 	}
 
+	/**
+	 * Sanitize HTML that are not added via Gutenberg.
+	 *
+	 * @param Document $dom  Document.
+	 *
+	 * @return void
+	 */
+	public function sanitize_raw_embeds( Document $dom ) {
+
+		$applicable_domains = [ 'youtu.be', 'youtube.com', 'youtube-nocookie.com' ];
+
+		$query_segments = array_map(
+			static function ( $domain ) {
+
+				return sprintf(
+					'starts-with( @src, "https://www.%1$s/" ) or starts-with( @src, "https://%1$s/" ) or starts-with( @src, "http://www.%1$s/" ) or starts-with( @src, "http://%1$s/" )',
+					$domain
+				);
+			},
+			$applicable_domains
+		);
+
+		$query = implode( ' or ', $query_segments );
+
+		$nodes = $dom->xpath->query( sprintf( '//iframe[ %s ]', $query ) );
+
+		/** @var Element $node */
+		foreach ( $nodes as $node ) {
+
+			$html             = $dom->saveHTML( $node );
+			$url              = $node->getAttribute( 'src' );
+			$amp_youtube_node = $this->process_embed( $dom, $html, $url );
+
+			if ( ! empty( $amp_youtube_node ) ) {
+				$node->parentNode->replaceChild( $amp_youtube_node, $node );
+			}
+		}
+
+	}
+
+	/**
+	 * To AMP youtube component from DOM Document.
+	 *
+	 * @param Document $dom  Document DOM.
+	 * @param string   $html HTML markup of youtube iframe.
+	 * @param string   $url  Youtube URL.
+	 *
+	 * @return DOMElement|false DOMElement on success, Otherwise false.
+	 */
+	public function process_embed( Document $dom, $html, $url ) {
+
+		$id = $this->get_video_id_from_url( $url );
+
+		if ( ! $id ) {
+			return false;
+		}
+
+		$args = $this->parse_props( $html, $url, $id );
+		if ( empty( $args ) ) {
+			return false;
+		}
+
+		$args['video_id'] = $id;
+
+		$attributes = $this->prepare_attributes( $args, $url );
+
+		if ( empty( $attributes['data-videoid'] ) ) {
+			return AMP_DOM_Utils::create_node(
+				$dom,
+				'a',
+				[
+					'href'  => esc_url_raw( $url ),
+					'class' => 'amp-wp-embed-fallback',
+				]
+			);
+		}
+
+		return AMP_DOM_Utils::create_node(
+			$dom,
+			'amp-youtube',
+			$attributes
+		);
```

Comment:
Don't forget the placeholder element:

```suggestion
		$placeholder_node = AMP_DOM_Utils::create_node(
			$dom,
			Tag::A,
			[
				Attribute::PLACEHOLDER => '',
				Attribute::HREF => esc_url_raw( $url ),
			]
		);

		$amp_component = AMP_DOM_Utils::create_node(
			$dom,
			'amp-youtube',
			$attributes
		);
		$amp_component->appendChild( $placeholder_node );

		return $amp_component;
```